cmd/axelard/cmd/vald/evm/rpc/rpcClient.go

```diff
@@ -2,37 +2,100 @@ package rpc
 
 import (
 	"context"
+	"math/big"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/types"
 	evmClient "github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/rpc"
 )
 
 //go:generate moq -out ./mock/rpcClient.go -pkg mock . Client
 
+const codeMethodNotFound = -32601
+
 // Client provides calls to an EVM RPC endpoint
 type Client interface {
 	BlockNumber(ctx context.Context) (uint64, error)
 	TransactionByHash(ctx context.Context, hash common.Hash) (tx *types.Transaction, isPending bool, err error)
 	TransactionReceipt(ctx context.Context, txHash common.Hash) (*types.Receipt, error)
+	BlockByNumber(ctx context.Context, number *big.Int) (*types.Block, error)
+	Close()
+
+	ChainGetFinalizedHead(ctx context.Context) (common.Hash, error)
+	ChainGetHeader(ctx context.Context, hash common.Hash) (*MoonbeamHeader, error)
+	IsMoonbeam() bool
```

Comment:
```suggestion
	BlockByNumber(ctx context.Context, number *big.Int) (*types.Block, error)
	Close()

type MoonbeamClient interface {
	Client
	ChainGetFinalizedHead(ctx context.Context) (common.Hash, error)
	ChainGetHeader(ctx context.Context, hash common.Hash) (*MoonbeamHeader, error)
```

And instead of an `isMoonbeam` function, you will however have to switch on the network name that is given by the config file to know which constructor to use